src/lib/SmartTokens/Contracts/Issuance.hs

```diff
@@ -109,12 +111,12 @@ mkProgrammableLogicMinting = plam $ \programmableLogicBase nodeCS mintingLogicCr
           # plam (pfstBuiltin #)
           # pto (pfromData infoF.wdrl)
 
-  pmatch red $ \case
+  pmatch (pfromData red) $ \case
```

Comment:
Because it is data encoded we need to use fromData to apply pasInt to get the underlying integer to branch on.